src/account/Account.ts

```diff
@@ -222,19 +216,26 @@ export abstract class Account {
   }
 
   /**
-   * Sign a message using the available signing capabilities.
-   * @param message the signing message, as binary input
-   * @return the AccountAuthenticator containing the signature, together with the account's public key
+   * Sign a transaction using the available signing capabilities.
+   * @param transaction the raw transaction
+   * @return the AccountAuthenticator containing the signature of the transaction, together with the account's public key
    */
-  abstract signWithAuthenticator(message: HexInput): AccountAuthenticator;
+  abstract signWithAuthenticator(transaction: AnyRawTransaction): AccountAuthenticator;
 
   /**
    * Sign the given message with the private key.
    * @param message in HexInput format
-   * @returns AccountSignature
+   * @returns Signature
    */
   abstract sign(message: HexInput): Signature;
 
+  /**
+   * Sign the given transaction.
+   * @param message in HexInput format
+   * @returns Signature
+   */
+  abstract signTransaction(transaction: AnyRawTransaction): Signature;
```

Comment:
is there a reason we need this function? when building the sdk, we intentionally decided to return  `AccountAuthenticator` from a sign functionality as it makes more sense and help with dev exp cc @hardsetting

Reply:
Right, we still maintain `signWithAuthenticator` so I'm not taking anything away.  I add `signTransaction` for the ease of implementing signWithAuthenticator in the MultiKeyAccount.  I just think the decomposition makes the code more readable and the abstract class easier to implement.  I'd be totally fine making it private if you feel it would be misused by devs or confuse them.  But I figured why not *shrug*.  lmk what you prefer.